### DISCUSSION TITLE
Installing Camera - Error During Setup

### DISCUSSION THREAD

**Discussion by schelli1** [paukstelis/octoprint_deploy#146] [Q&A]
Hi loving the software! I am trying to install a camera but am getting the following error during the install. Any assistance would be much appreciated thank you!

<img width="552" alt="Screenshot 2024-01-21 at 2 23 26 pm" src="https://github.com/paukstelis/octoprint_deploy/assets/109404975/b95d44b5-0668-46c1-98b0-43910d895bca">

**Comment by paukstelis** [comment]
In this case not all packages were properly downloaded during the system prepare.
You can try to reinstall all the needed packages with: `sudo octoprint_deploy/octoprint_deploy.sh f deb_packages`
Then you can try the streamer installation again.

**Reply by schelli1** [reply]
Hi, thank you for your help. I couldn't, unfortunately, get it to work, I ran that script you sent, uninstalled the camera then reinstalled it. The pi does pick up the USB connection so not sure what is going on. The generated log is bellow.


octoprint_deploy diagnostic information. Please provide ALL output for support help
**************************************
/etc/octoprint_deploy
**************************************
octoexec: /home/pi/oprint/bin/octoprint
octopip: /home/pi/oprint/bin/pip
haproxy: true
octopi: true
streamer: ustreamer
**************************************
/etc/octoprint_instances
**************************************
instance:octoprint port:5000 udev:false
instance:kobra_max port:5001 udev:true
instance:ender3max port:5002 udev:true
**************************************
/etc/octoprint_cameras
**************************************
camera:cam_kobra_max port:8001 udev:true
**************************************
/etc/udev/rules.d/99-octoprint.rules
**************************************
KERNELS=="1-1.1.2:1.0",SUBSYSTEM=="tty",SYMLINK+="octo_kobra_max"
KERNELS=="1-1.3:1.0",SUBSYSTEM=="tty",SYMLINK+="octo_ender3max"
SUBSYSTEM=="video4linux", ATTRS{serial}=="YG_U600D.4653_4K.2103121705", ATTR{index}=="0", SYMLINK+="cam_kobra_max"
lrwxrwxrwx 1 root root 7 Jan 21 20:59 /dev/octo_ender3max -> ttyUSB1
lrwxrwxrwx 1 root root 7 Jan 21 20:52 /dev/octo_kobra_max -> ttyUSB0
lrwxrwxrwx 1 root root 6 Jan 21 20:52 /dev/cam_kobra_max -> video1
**************************************
● octoprint.service - The snappy web interface for your 3D printer
     Loaded: loaded (/etc/systemd/system/octoprint.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2024-01-20 22:12:15 GMT; 23h ago
   Main PID: 399 (octoprint)
      Tasks: 217 (limit: 1472)
        CPU: 18min 56.239s
     CGroup: /system.slice/octoprint.service
             └─399 /home/pi/oprint/bin/python /home/pi/oprint/bin/octoprint serve --host=127.0.0.1 --port=5000

Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,110 - octoprint.util.comm - INFO - Serial detection: Trying port /dev/ttyUSB1, baudrate 250000
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,114 - octoprint.util.comm - INFO - Serial detection: Handshake attempt #1 with timeout 2.0s
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,118 - octoprint.util.comm - INFO - M110 detected, setting current line number to 0
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,129 - octoprint.util.comm - INFO - Changing monitoring state from "Detecting serial connection" to "Operational"
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,140 - octoprint.util.comm - INFO - M110 detected, setting current line number to 0
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,210 - octoprint.util.comm - INFO - Printer reports firmware name "Marlin bugfix-2.1.x-mfs (Jan 20 2024 10:18:08)"
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,213 - octoprint.plugins.firmware_check - INFO - Your printer's firmware is a nightly build of Marlin by the Marlin Firmware Service (build date 20240120). It might be more unstable than a release version and should be kept up-to-date.. More information at https://faq.octoprint.org/warning-firmware-development-mfs
Jan 21 21:00:06 octopi octoprint[399]: 2024-01-21 21:00:06,272 - octoprint.util.comm - INFO - Firmware states that it supports temperature autoreporting
Jan 21 21:11:07 octopi octoprint[399]: 2024-01-21 21:11:07,279 - octoprint.server.util.sockjs - INFO - Client connection closed: ::ffff:10.0.0.27
Jan 21 21:12:55 octopi octoprint[399]: 2024-01-21 21:12:55,738 - octoprint.server.heartbeat - INFO - Server heartbeat <3
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
● kobra_max.service - The snappy web interface for your 3D printer
     Loaded: loaded (/etc/systemd/system/kobra_max.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2024-01-21 20:52:39 GMT; 22min ago
   Main PID: 14073 (octoprint)
      Tasks: 28 (limit: 1472)
        CPU: 2min 16.316s
     CGroup: /system.slice/kobra_max.service
             └─14073 /home/pi/oprint/bin/python /home/pi/oprint/bin/octoprint serve --config=/home/pi/.kobra_max/config.yaml --basedir=/home/pi/.kobra_max --port=5001

Jan 21 21:00:51 octopi octoprint[14073]: 2024-01-21 21:00:51,885 - octoprint.server.util.sockjs - INFO - New connection from client: ::ffff:10.0.0.27
Jan 21 21:00:51 octopi octoprint[14073]: 2024-01-21 21:00:51,934 - octoprint.server.util.flask - INFO - Passively logging in user muir from ::ffff:10.0.0.27
Jan 21 21:00:51 octopi octoprint[14073]: 2024-01-21 21:00:51,936 - octoprint.access.users - INFO - Logged in user: muir
Jan 21 21:00:51 octopi octoprint[14073]: 2024-01-21 21:00:51,981 - octoprint.server.util.sockjs - INFO - User muir logged in on the socket from client ::ffff:10.0.0.27
Jan 21 21:00:54 octopi octoprint[14073]: 2024-01-21 21:00:54,430 - octoprint.server.util.sockjs - INFO - Client connection closed: ::ffff:10.0.0.27
Jan 21 21:00:54 octopi octoprint[14073]: 2024-01-21 21:00:54,488 - octoprint.server.util.sockjs - INFO - New connection from client: ::ffff:10.0.0.27
Jan 21 21:00:54 octopi octoprint[14073]: 2024-01-21 21:00:54,550 - octoprint.server.util.flask - INFO - Passively logging in user muir from ::ffff:10.0.0.27
Jan 21 21:00:54 octopi octoprint[14073]: 2024-01-21 21:00:54,551 - octoprint.access.users - INFO - Logged in user: muir
Jan 21 21:00:54 octopi octoprint[14073]: 2024-01-21 21:00:54,723 - octoprint.server.util.sockjs - INFO - User muir logged in on the socket from client ::ffff:10.0.0.27
Jan 21 21:07:54 octopi octoprint[14073]: 2024-01-21 21:07:54,003 - octoprint.server.heartbeat - INFO - Server heartbeat <3
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
● ender3max.service - The snappy web interface for your 3D printer
     Loaded: loaded (/etc/systemd/system/ender3max.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2024-01-20 22:12:15 GMT; 23h ago
   Main PID: 396 (octoprint)
      Tasks: 238 (limit: 1472)
        CPU: 40min 57.760s
     CGroup: /system.slice/ender3max.service
             └─396 /home/pi/oprint/bin/python /home/pi/oprint/bin/octoprint serve --config=/home/pi/.ender3max/config.yaml --basedir=/home/pi/.ender3max --port=5002

Jan 21 21:12:24 octopi octoprint[396]:   File "/usr/lib/python3.9/concurrent/futures/thread.py", line 170, in submit
Jan 21 21:12:24 octopi octoprint[396]:     self._adjust_thread_count()
Jan 21 21:12:24 octopi octoprint[396]:   File "/usr/lib/python3.9/concurrent/futures/thread.py", line 193, in _adjust_thread_count
Jan 21 21:12:24 octopi octoprint[396]:     t.start()
Jan 21 21:12:24 octopi octoprint[396]:   File "/home/pi/oprint/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 56, in sentry_start
Jan 21 21:12:24 octopi octoprint[396]:     return old_start(self, *a, **kw)
Jan 21 21:12:24 octopi octoprint[396]:   File "/usr/lib/python3.9/threading.py", line 874, in start
Jan 21 21:12:24 octopi octoprint[396]:     _start_new_thread(self._bootstrap, ())
Jan 21 21:12:24 octopi octoprint[396]: RuntimeError: can't start new thread
Jan 21 21:12:55 octopi octoprint[396]: 2024-01-21 21:12:55,762 - octoprint.server.heartbeat - INFO - Server heartbeat <3
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
● cam_kobra_max.service - the OctoPi(buntu) ustreamer daemon with the user specified config
     Loaded: loaded (/etc/systemd/system/cam_kobra_max.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Sun 2024-01-21 20:52:39 GMT; 22min ago
   Main PID: 14095 (code=exited, status=203/EXEC)
        CPU: 4ms

Jan 21 20:52:39 octopi systemd[14095]: cam_kobra_max.service: Failed at step EXEC spawning /home/pi/ustreamer/ustreamer: No such file or directory
Jan 21 20:52:39 octopi systemd[1]: Started the OctoPi(buntu) ustreamer daemon with the user specified config.
Jan 21 20:52:39 octopi systemd[1]: cam_kobra_max.service: Main process exited, code=exited, status=203/EXEC
Jan 21 20:52:39 octopi systemd[1]: cam_kobra_max.service: Failed with result 'exit-code'.
**************************************
● haproxy.service - HAProxy Load Balancer
     Loaded: loaded (/lib/systemd/system/haproxy.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2024-01-21 20:52:39 GMT; 22min ago
       Docs: man:haproxy(1)
             file:/usr/share/doc/haproxy/configuration.txt.gz
   Main PID: 14087 (haproxy)
      Tasks: 5 (limit: 1472)
        CPU: 3.007s
     CGroup: /system.slice/haproxy.service
             ├─14087 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -S /run/haproxy-master.sock
             └─14089 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -S /run/haproxy-master.sock

Jan 21 21:12:24 octopi haproxy[14089]: ::ffff:10.0.0.22:58342 [21/Jan/2024:21:12:24.708] public ender3max/octoprint1 3/0/1/213/217 200 2475 - - ---- 11/11/3/3/0 0/0 "GET /ender3max/api/access/users HTTP/1.1"
Jan 21 21:12:24 octopi haproxy[14089]: ::ffff:10.0.0.22:58344 [21/Jan/2024:21:12:24.654] public ender3max/octoprint1 0/0/1/324/325 200 793 - - ---- 11/11/2/2/0 0/0 "POST /ender3max/plugin/PrintTimeGenius/print_history HTTP/1.1"
Jan 21 21:12:25 octopi haproxy[14089]: ::ffff:10.0.0.22:58345 [21/Jan/2024:21:12:25.235] public ender3max/octoprint1 48/0/0/46/94 304 870 - - ---- 11/11/2/2/0 0/0 "GET /ender3max/plugin/gcodeviewer/static/js/lib/pako.js HTTP/1.1"
Jan 21 21:12:43 octopi haproxy[14089]: ::ffff:10.0.0.27:64610 [21/Jan/2024:21:12:40.903] public cam_kobra_max/webcam1 0/3005/-1/-1/3005 503 222 - - SC-- 11/11/0/0/3 0/0 "GET /cam_kobra_max/?action=stream HTTP/1.1"
Jan 21 21:14:03 octopi haproxy[14089]: ::ffff:10.0.0.68:53752 [21/Jan/2024:21:14:03.586] public octoprint/octoprint1 0/0/0/27/27 200 933 - - ---- 12/12/0/0/0 0/0 "GET /plugin/discovery/discovery.xml HTTP/1.1"
Jan 21 21:14:03 octopi haproxy[14089]: ::ffff:10.0.0.68:53764 [21/Jan/2024:21:14:03.900] public octoprint/octoprint1 0/0/0/26/26 200 933 - - ---- 12/12/0/0/0 0/0 "GET /plugin/discovery/discovery.xml HTTP/1.1"
Jan 21 21:14:07 octopi haproxy[14089]: ::ffff:10.0.0.27:64610 [21/Jan/2024:21:14:04.520] public cam_kobra_max/webcam1 0/3005/-1/-1/3005 503 222 - - SC-- 11/11/0/0/3 0/0 "GET /cam_kobra_max/?action=stream HTTP/1.1"
Jan 21 21:14:51 octopi haproxy[14089]: ::ffff:10.0.0.27:65072 [21/Jan/2024:21:14:51.871] public octoprint/octoprint1 0/0/0/61/61 200 933 - - ---- 12/12/0/0/0 0/0 "GET /plugin/discovery/discovery.xml HTTP/1.1"
Jan 21 21:15:05 octopi haproxy[14089]: ::ffff:10.0.0.43:62732 [21/Jan/2024:21:15:04.967] public octoprint/octoprint1 0/0/1/64/65 200 933 - - ---- 12/12/0/0/0 0/0 "GET /plugin/discovery/discovery.xml HTTP/1.1"
Jan 21 21:15:17 octopi haproxy[14089]: ::ffff:10.0.0.43:62742 [21/Jan/2024:21:15:16.950] public octoprint/octoprint1 0/0/1/64/65 200 933 - - ---- 12/12/0/0/0 0/0 "GET /plugin/discovery/discovery.xml HTTP/1.1"

**Comment by paukstelis** [comment]
because ustreamer is still not installed.

what is the output from the `sudo octoprint_deploy/octoprint_deploy.sh f deb_packages` ?

**Reply by schelli1** [reply]
pi@octopi:~ $ sudo octoprint_deploy/octoprint_deploy.sh f deb_packages
[sudo] password for pi:
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
build-essential is already the newest version (12.9).
g++ is already the newest version (4:10.2.1-1+rpi1).
g++ set to manually installed.
gcc is already the newest version (4:10.2.1-1+rpi1).
gcc set to manually installed.
libffi-dev is already the newest version (3.3-6).
libjpeg62-turbo-dev is already the newest version (1:2.0.6-4).
make is already the newest version (4.3-4.1).
make set to manually installed.
pkg-config is already the newest version (0.29.2-1).
python-is-python3 is already the newest version (3.9.2-1).
python3-dev is already the newest version (3.9.2-3).
python3-setuptools is already the newest version (52.0.0-4).
python3-setuptools set to manually installed.
rsync is already the newest version (3.2.3-4+deb11u1).
ssh is already the newest version (1:8.4p1-5+deb11u1).
v4l-utils is already the newest version (1.20.0-2).
xxd is already the newest version (2:8.2.2434-3+deb11u1).
cmake is already the newest version (3.18.4-2+rpt1+rpi1+deb11u1).
python3-pip is already the newest version (20.3.4-4+rpt1+deb11u1).
The following additional packages will be installed:
  libavcodec58 libavdevice58 libavfilter7 libavformat58 libavresample4
  libavutil56 libbasicusageenvironment1 libbsd0 libevent-core-2.1-7
  libevent-extra-2.1-7 libevent-openssl-2.1-7 libevent-pthreads-2.1-7
  libgroupsock8 liblivemedia77 libmd-dev libpostproc55 libssl1.1
  libswresample-dev libswresample3 libswscale5 libusageenvironment3
  libwebpmux3
Suggested packages:
  ffmpeg-doc vim-haproxy haproxy-doc libssl-doc libyaml-doc
The following NEW packages will be installed:
  libavcodec-dev libavformat-dev libavutil-dev libbasicusageenvironment1
  libbsd-dev libcamera-dev libevent-core-2.1-7 libevent-dev
  libevent-extra-2.1-7 libevent-openssl-2.1-7 libevent-pthreads-2.1-7
  libgroupsock8 libjpeg-dev liblivemedia-dev liblivemedia77 libmd-dev
  libswresample-dev libusageenvironment3 libyaml-dev python3-venv
  python3.9-venv uuid-runtime virtualenv
The following packages will be upgraded:
  ffmpeg haproxy libavcodec58 libavdevice58 libavfilter7 libavformat58
  libavresample4 libavutil56 libbsd0 libpostproc55 libssl-dev libssl1.1
  libswresample3 libswscale5 libwebpmux3
15 upgraded, 23 newly installed, 0 to remove and 132 not upgraded.
Need to get 3,349 kB/34.6 MB of archives.
After this operation, 50.9 MB of additional disk space will be used.
Err:2 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf haproxy armhf                                                                                                                                                                                                                                              2.2.9-2+rpi1+deb11u5
  404  Not Found [IP: 93.93.128.193 80]
Err:1 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf libssl-dev armhf 1.1.1w-0+deb11u1
  Connection failed [IP: 150.101.135.3 80]
E: Failed to fetch http://mirror.internode.on.net/pub/raspbian/raspbian/pool/main/o/openssl/libssl-dev_1.1.1w-0+deb11u1_armhf.deb  Connection failed [IP: 150.101.135.3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/h/haproxy/haproxy_2.2.9-2%2brpi1%2bdeb11u5_armhf.deb  404  Not Found [IP: 93.93.128.193 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package 'brltty' is not installed, so not removed
0 upgraded, 0 newly installed, 0 to remove and 147 not upgraded.

**Comment by paukstelis** [comment]
This seems to be an issue with OctoPi and how packages have changed.
try:
`sudo apt-get update`
`sudo octoprint_deploy/octoprint_deploy.sh f deb_packages`

That should get everything downloaded and installed.
Then do the "Change Streamer" command from octoprint_deploy menu. You shouldn't have to remove/reinstall camera, just do a `sudo systemctl restart cam_kobra_max`

**Reply by schelli1** [reply]
Once again thank you and sorry for the trouble. Still getting an error.

pi@octopi:~ $ sudo apt-get update
[sudo] password for pi:
Get:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease [15.0 kB]
Get:2 http://archive.raspberrypi.org/debian bullseye InRelease [23.6 kB]
Get:3 http://archive.raspberrypi.org/debian bullseye/main armhf Packages [313 kB
Get:4 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf Packages [13.2 MB]
Fetched 13.6 MB in 48s (285 kB/s)
Reading package lists... Done
N: Repository 'http://archive.raspberrypi.org/debian bullseye InRelease' changed its 'Suite' value from 'stable' to 'oldstable'
pi@octopi:~ $ sudo octoprint_deploy/octoprint_deploy.sh f deb_packages
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
build-essential is already the newest version (12.9).
g++ is already the newest version (4:10.2.1-1+rpi1).
g++ set to manually installed.
gcc is already the newest version (4:10.2.1-1+rpi1).
gcc set to manually installed.
libffi-dev is already the newest version (3.3-6).
libjpeg62-turbo-dev is already the newest version (1:2.0.6-4).
make is already the newest version (4.3-4.1).
make set to manually installed.
pkg-config is already the newest version (0.29.2-1).
python-is-python3 is already the newest version (3.9.2-1).
python3-dev is already the newest version (3.9.2-3).
python3-setuptools is already the newest version (52.0.0-4).
python3-setuptools set to manually installed.
rsync is already the newest version (3.2.3-4+deb11u1).
v4l-utils is already the newest version (1.20.0-2).
xxd is already the newest version (2:8.2.2434-3+deb11u1).
cmake is already the newest version (3.18.4-2+rpt1+rpi1+deb11u1).
python3-pip is already the newest version (20.3.4-4+rpt1+deb11u1).
The following additional packages will be installed:
  libavcodec58 libavdevice58 libavfilter7 libavformat58 libavresample4
  libavutil56 libbasicusageenvironment1 libbsd0 libevent-core-2.1-7
  libevent-extra-2.1-7 libevent-openssl-2.1-7 libevent-pthreads-2.1-7
  libgroupsock8 liblivemedia77 libmd-dev libpostproc55 libssl1.1
  libswresample-dev libswresample3 libswscale5 libusageenvironment3
  libwebpmux3 openssh-client openssh-server openssh-sftp-server
Suggested packages:
  ffmpeg-doc vim-haproxy haproxy-doc libssl-doc libyaml-doc keychain
  libpam-ssh monkeysphere ssh-askpass molly-guard ufw
The following NEW packages will be installed:
  libavcodec-dev libavformat-dev libavutil-dev libbasicusageenvironment1
  libbsd-dev libcamera-dev libevent-core-2.1-7 libevent-dev
  libevent-extra-2.1-7 libevent-openssl-2.1-7 libevent-pthreads-2.1-7
  libgroupsock8 libjpeg-dev liblivemedia-dev liblivemedia77 libmd-dev
  libswresample-dev libusageenvironment3 libyaml-dev python3-venv
  python3.9-venv uuid-runtime virtualenv
The following packages will be upgraded:
  ffmpeg haproxy libavcodec58 libavdevice58 libavfilter7 libavformat58
  libavresample4 libavutil56 libbsd0 libpostproc55 libssl-dev libssl1.1
  libswresample3 libswscale5 libwebpmux3 openssh-client openssh-server
  openssh-sftp-server ssh
19 upgraded, 23 newly installed, 0 to remove and 135 not upgraded.
Need to get 6,071 kB/36.0 MB of archives.
After this operation, 51.0 MB of additional disk space will be used.
Get:1 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf haproxy armhf 2.2.9-2+rpi1+deb11u6 [1,763 kB]
Get:2 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf openssh-sftp-server armhf 1:8.4p1-5+deb11u3 [42.8 kB]
Get:3 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf openssh-server armhf 1:8.4p1-5+deb11u3 [319 kB]
Get:4 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf openssh-client armhf 1:8.4p1-5+deb11u3 [796 kB]
Get:5 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf ssh all 1:8.4p1-5+deb11u3 [252 kB]
Err:6 http://archive.raspberrypi.org/debian bullseye/main armhf libssl-dev armhf 1.1.1w-0+deb11u1+rpt1
  Connection failed [IP: 46.235.231.111 80]
Get:7 http://archive.raspberrypi.org/debian bullseye/main armhf libssl1.1 armhf 1.1.1w-0+deb11u1+rpt1 [1,298 kB]
Fetched 4,470 kB in 1min 8s (65.9 kB/s)
E: Failed to fetch http://archive.raspberrypi.org/debian/pool/main/o/openssl/libssl-dev_1.1.1w-0%2bdeb11u1%2brpt1_armhf.deb  Connection failed [IP: 46.235.231.111 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package 'brltty' is not installed, so not removed
0 upgraded, 0 newly installed, 0 to remove and 154 not upgraded.


*************************
octoprint_deploy 1.0.7
*************************

1) Add instance     3) Add Camera       5) Utilities        7) Update
2) Delete instance  4) Delete Camera    6) Backup Menu      8) Quit
Select operation: 5


1) Instance Status    4) Share Uploads      7) Udev Menu
2) USB Port Testing   5) Change Streamer    8) Diagnostic Output
3) Sync Users         6) Set Global Config  9) Quit
Select an option: 5

This allows you to change the default webcam streamer.
Please note, it DOES NOT change the streamer for existing cameras.
You will need to delete and reinstall those cameras for changes to take effect.

1) ustreamer (recommended)
2) None/Skip
Which video streamer you would like to install?: 1
Cloning into 'ustreamer'...
remote: Enumerating objects: 179, done.
remote: Counting objects: 100% (179/179), done.
remote: Compressing objects: 100% (130/130), done.
remote: Total 179 (delta 61), reused 108 (delta 37), pack-reused 0
Receiving objects: 100% (179/179), 179.06 KiB | 1.38 MiB/s, done.
Resolving deltas: 100% (61/61), done.
In file included from ustreamer/http/bev.c:23:
ustreamer/http/bev.h:28:10: fatal error: event2/util.h: No such file or directory
   28 | #include <event2/util.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:100: build/ustreamer/http/bev.o] Error 1
make[1]: *** [Makefile:38: apps] Error 2
make: *** [Makefile:28: all] Error 2
WARNING! WARNING! WARNING!
Streamer has not been installed correctly.
Try installation again? [y/n]: N
pi@octopi:~ $ sudo systemctl restart cam_kobra_max
pi@octopi:~ $ sudo octoprint_deploy/octoprint_deploy.sh f deb_packages
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
build-essential is already the newest version (12.9).
g++ is already the newest version (4:10.2.1-1+rpi1).
g++ set to manually installed.
gcc is already the newest version (4:10.2.1-1+rpi1).
gcc set to manually installed.
libffi-dev is already the newest version (3.3-6).
libjpeg62-turbo-dev is already the newest version (1:2.0.6-4).
make is already the newest version (4.3-4.1).
make set to manually installed.
pkg-config is already the newest version (0.29.2-1).
python-is-python3 is already the newest version (3.9.2-1).
python3-dev is already the newest version (3.9.2-3).
python3-setuptools is already the newest version (52.0.0-4).
python3-setuptools set to manually installed.
rsync is already the newest version (3.2.3-4+deb11u1).
v4l-utils is already the newest version (1.20.0-2).
xxd is already the newest version (2:8.2.2434-3+deb11u1).
cmake is already the newest version (3.18.4-2+rpt1+rpi1+deb11u1).
python3-pip is already the newest version (20.3.4-4+rpt1+deb11u1).
The following additional packages will be installed:
  libavcodec58 libavdevice58 libavfilter7 libavformat58 libavresample4
  libavutil56 libbasicusageenvironment1 libbsd0 libevent-core-2.1-7
  libevent-extra-2.1-7 libevent-openssl-2.1-7 libevent-pthreads-2.1-7
  libgroupsock8 liblivemedia77 libmd-dev libpostproc55 libssl1.1
  libswresample-dev libswresample3 libswscale5 libusageenvironment3
  libwebpmux3 openssh-client openssh-server openssh-sftp-server
Suggested packages:
  ffmpeg-doc vim-haproxy haproxy-doc libssl-doc libyaml-doc keychain
  libpam-ssh monkeysphere ssh-askpass molly-guard ufw
The following NEW packages will be installed:
  libavcodec-dev libavformat-dev libavutil-dev libbasicusageenvironment1
  libbsd-dev libcamera-dev libevent-core-2.1-7 libevent-dev
  libevent-extra-2.1-7 libevent-openssl-2.1-7 libevent-pthreads-2.1-7
  libgroupsock8 libjpeg-dev liblivemedia-dev liblivemedia77 libmd-dev
  libswresample-dev libusageenvironment3 libyaml-dev python3-venv
  python3.9-venv uuid-runtime virtualenv
The following packages will be upgraded:
  ffmpeg haproxy libavcodec58 libavdevice58 libavfilter7 libavformat58
  libavresample4 libavutil56 libbsd0 libpostproc55 libssl-dev libssl1.1
  libswresample3 libswscale5 libwebpmux3 openssh-client openssh-server
  openssh-sftp-server ssh
19 upgraded, 23 newly installed, 0 to remove and 135 not upgraded.
Need to get 1,601 kB/36.0 MB of archives.
After this operation, 51.0 MB of additional disk space will be used.
Err:1 http://archive.raspberrypi.org/debian bullseye/main armhf libssl-dev armhf 1.1.1w-0+deb11u1+rpt1
  Connection failed [IP: 46.235.231.151 80]
E: Failed to fetch http://archive.raspberrypi.org/debian/pool/main/o/openssl/libssl-dev_1.1.1w-0%2bdeb11u1%2brpt1_armhf.deb  Connection failed [IP: 46.235.231.151 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package 'brltty' is not installed, so not removed
0 upgraded, 0 newly installed, 0 to remove and 154 not upgraded.


*************************
octoprint_deploy 1.0.7
*************************

1) Add instance     3) Add Camera       5) Utilities        7) Update
2) Delete instance  4) Delete Camera    6) Backup Menu      8) Quit
Select operation: 5


1) Instance Status    4) Share Uploads      7) Udev Menu
2) USB Port Testing   5) Change Streamer    8) Diagnostic Output
3) Sync Users         6) Set Global Config  9) Quit
Select an option: 5

This allows you to change the default webcam streamer.
Please note, it DOES NOT change the streamer for existing cameras.
You will need to delete and reinstall those cameras for changes to take effect.

1) ustreamer (recommended)
2) None/Skip
Which video streamer you would like to install?: 1
Cloning into 'ustreamer'...
remote: Enumerating objects: 179, done.
remote: Counting objects: 100% (179/179), done.
remote: Compressing objects: 100% (130/130), done.
remote: Total 179 (delta 61), reused 108 (delta 37), pack-reused 0
Receiving objects: 100% (179/179), 179.06 KiB | 1.54 MiB/s, done.
Resolving deltas: 100% (61/61), done.
In file included from ustreamer/http/bev.c:23:
ustreamer/http/bev.h:28:10: fatal error: event2/util.h: No such file or directory
   28 | #include <event2/util.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:100: build/ustreamer/http/bev.o] Error 1
make[1]: *** [Makefile:38: apps] Error 2
make: *** [Makefile:28: all] Error 2
WARNING! WARNING! WARNING!
Streamer has not been installed correctly.
Try installation again? [y/n]:
